pkg/asset/agent/image/unconfigured_ignition.go

```diff
@@ -150,20 +158,45 @@ func (a *UnconfiguredIgnition) Generate(_ context.Context, dependencies asset.Pa
 		InfraEnvID:                infraEnvID,
 		OSImage:                   osImage,
 		Proxy:                     infraEnv.Spec.Proxy,
-		ConfigImageFiles:          configImageFiles,
 	}
 
-	err = bootstrap.AddStorageFiles(&config, "/", "agent/files", agentTemplateData)
-	if err != nil {
-		return err
-	}
+	enabledServices := getDefaultEnabledServices()
 
-	err = addBootstrapScripts(&config, clusterImageSetAsset.Config.Spec.ReleaseImage)
-	if err != nil {
-		return err
+	switch agentWorkflow.Workflow {
+	case workflow.AgentWorkflowTypeInstall:
+		agentTemplateData.ConfigImageFiles = strings.Join(GetConfigImageFiles(), ",")
+
+		// Enable the agent-check-config-image.service for the current workflow.
+		enabledServices = append(enabledServices, "agent-check-config-image.service")
+
+	case workflow.AgentWorkflowTypeInstallInteractiveDisconnected:
+		// Add the rendezvous host file. Agent TUI will interact with that file in case
+		// the rendezvous IP wasn't previously configured, by managing it as a template file.
+		rendezvousIP := "{{.RendezvousIP}}"
+		if agentConfig.Config != nil {
+			rendezvousIP = agentConfig.Config.RendezvousIP
+		}
+		rendezvousHostFile := ignition.FileFromString(rendezvousHostEnvPath, "root", 0644,
+			getRendezvousHostEnv("http", rendezvousIP, "", "", agentWorkflow.Workflow))
+		config.Storage.Files = append(config.Storage.Files, rendezvousHostFile)
+
+		// Explicitly disable the load-config-iso service, not required in the current flow
+		// (even though disabled by default, the udev rule may require it).
+		config.Storage.Files = append(config.Storage.Files, ignition.FileFromString("/etc/assisted/no-config-image", "root", 0644, ""))
+
+		// Enable the UI service.
+		enabledServices = append(enabledServices, "agent-start-ui.service")
+		interactiveUIFile := ignition.FileFromString("/etc/assisted/interactive-ui", "root", 0644, "")
```

Comment:
Are there two files? /etc/assisted/interactive-ui and /etc/assisted/interactive-ui.env referenced in agent-register-infraenv.service and agent-register-cluster.service?

Reply:
It was a typo, switching to plain empty sentinel file `/etc/assisted/interactive-ui`